bug59153_CSVDataSetFilesExceptions

ra0077 opened the pull request:
More clear message for CSV Data Set with 
filename empty
wrong filename (not readable, not exist)

I don't have made formating to help reviewing

If anybody is ok, I will format it

Antonio

ra0077 committed:
bug59153_CSVDataSetFilesExceptions

More clear message for CSV Data Set with 
filename empty
wrong filename (not readable, not exist)

I don't have made formating to help reviewing

If anybody is ok, I will format it

Antonio

vlsi commented on src/core/org/apache/jmeter/services/FileServer.java:
```diff
@@ -428,6 +429,9 @@ private BufferedReader createBufferedReader(FileEntry fileEntry) throws IOExcept
         }
         return new BufferedReader(isr);
+        } else {
+            throw new IllegalArgumentException("File "+ fileEntry.file.getName()+ " must exist and be readable");
```
Please move `else` branch to the start of the method.

ra0077 commented:
Ok

I will wait other review or a go before modify my PR to avoid problem like in my previous PR about this bug

Antonio

ra0077 committed:
bug59153_CSVDataSetFilesExceptions

    More clear message for CSV Data Set with
    filename empty
    wrong filename (not readable, not exist)
    New commit with formating and swap else branch

    Antonio

ra0077 commented:
New commit with formating and swap else branch

vlsi commented on src/core/org/apache/jmeter/services/FileServer.java:
```diff
@@ -418,16 +418,20 @@ private BufferedReader getReader(String alias, boolean recycle, boolean firstLin
-            isr = new InputStreamReader(fis, charsetName);
+        if (!fileEntry.file.exists() || !fileEntry.file.canRead() || !fileEntry.file.isFile()) {
+            throw new IllegalArgumentException("File "+ fileEntry.file.getName()+ " must exist and be readable");
         } else {
```
Please remove `else` and the following braces.
It would avoid indenting the following code.

ra0077 committed:
bug59153_CSVDataSetFilesExceptions

exists() removed and else removed

Antonio

vlsi commented on src/core/org/apache/jmeter/services/FileServer.java:
```diff
@@ -428,7 +431,7 @@ private BufferedReader createBufferedReader(FileEntry fileEntry) throws IOExcept
         }
         return new BufferedReader(isr);
-    }
+        }
```
Would you avoid adding extra spaces?

ra0077 committed:
bug59153_CSVDataSetFilesExceptions

Remove extra spaces

Antonio

ra0077 committed:
bug59153_CSVDataSetFilesExceptions
 
 Fix uniarty tests
 
 Antonio
